libraries/standard/http/include/http_client.h

```diff
@@ -215,18 +224,76 @@ typedef enum HTTPStatus
     HTTP_INSUFFICIENT_MEMORY,
 
     /**
-     * @brief Represents all errors not related to user-input or transport I/O, but
-     * errors internal to the implementation of the HTTP client library.
+     * @brief The server sent more headers than the configured
+     * #HTTP_MAX_RESPONSE_HEADERS_SIZE_BYTES.
      *
      * Functions that may return this value:
      * - #HTTPClient_Send
-     * - #HTTPClient_ReadHeader
      */
-    HTTP_INTERNAL_ERROR,
     HTTP_SECURITY_ALERT_RESPONSE_HEADERS_SIZE_LIMIT_EXCEEDED,
-    HTTP_SECURITY_ALERT_PARSER_INVALID_CHARACTER,
-    HTTP_SECURITY_ALERT_INVALID_CONTENT_LENGTH,
-    /* TODO: Add return codes as implementation continues. */
+
+    /**
+     * @brief The server data past a complete response that contained the
+     * "Connection: close" header.
+     *
+     * Functions that may return this value:
+     * - #HTTPClient_Send
+     */
+    HTTP_SECURITY_ALERT_EXTRANEOUS_RESPONSE_DATA,
+
+    /**
+     * @brief The server sent a chunk header that contained an invalid
+     * character.
+     *
+     * Functions that may return this value:
+     * - #HTTPClient_Send
+     */
+    HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_CHUNK_HEADER,
+
+    /**
+     * @brief The server sent a response with an invalid character in the
+     * HTTP protocol version.
+     *
+     * Functions that may return this value:
+     * - #HTTPClient_Send
+     */
+    HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_PROTOCOL_VERSION,
+
+    /**
+     * @brief The server sent a response with an invalid character in the
+     * HTTP status-code or the HTTP status code is out of range.
+     *
+     * Functions that may return this value:
+     * - #HTTPClient_Send
+     */
+    HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_STATUS_CODE,
+
+    /**
+     * @brief An invalid character was found in the HTTP response message.
+     *
+     * Functions that may return this value:
+     * - #HTTPClient_Send
+     */
+    HTTP_SECURITY_ALERT_MALFORMED_RESPONSE_INVALID_CHARACTER,
+
+    /**
+     * @brief An invalid character was found in the Content-Length length
```

Comment:
(Nit recommendation for use of _either_ in the sentence ): The server response _either_ contains an invalid character in the Content-Length header or contains the Content-Length header when it is unexpected to be present